Cannot compile network logging backend with IPv6-only

I would like to use the network logging backend from a firmware that has IPv4 disabled and only IPv6 enabled. But this leads to a compilation error.

```
<zephyrproject>/zephyr/subsys/logging/log_backend_net.c:138:27: error: 'struct net_if_ip' has no member named 'ipv4'; did you mean 'ipv6'?
  138 |   ipv4 = iface->config.ip.ipv4;
      |                           ^~~~
      |                           ipv6
```

This should be reproducible with any firmware that has:

```
CONFIG_NET_IPV6=y
# CONFIG_NET_IPV4 is not set
CONFIG_LOG_BACKEND_NET=y
```

The issue can be reproduced with the sample samples/net/syslog_net/, when in prj.conf IPv4 is disabled, see commit https://github.com/lemonbeat/zephyr/commit/2fc5b2c80dd984fde34f46f7ecbc1a1d7eb62dd3.

My firmware is based on zephyr revision a043d48c54726082521f3e335d807dc649cc02f3. It also happens on the current master 27ed1c3e138f755d19055a378382f767b5bd9ebd

**Expected behavior**
I would expect that logging to a syslog server works from devices with IPv4 disabled and IPv6 enabled.

--- a/samples/net/syslog_net/sample.yaml
+++ b/samples/net/syslog_net/sample.yaml
@@ -12,3 +12,16 @@ tests:
   sample.net.syslog.without_timefuncs:
     extra_configs:
       - CONFIG_NEWLIB_LIBC=n
+  sample.net.syslog.ipv4_only:
+    extra_configs:
+      - CONFIG_NET_IPV6=n
+      - CONFIG_NET_CONFIG_NEED_IPV6=n
+      - CONFIG_NET_CONFIG_MY_IPV6_ADDR=""
+      - CONFIG_NET_CONFIG_PEER_IPV6_ADDR=""
+      - CONFIG_LOG_BACKEND_NET_SERVER="192.0.2.1:514"
+  sample.net.syslog.ipv6_only:
+    extra_configs:
+      - CONFIG_NET_IPV4=n
+      - CONFIG_NET_CONFIG_NEED_IPV4=n
+      - CONFIG_NET_CONFIG_MY_IPV4_ADDR=""
+      - CONFIG_NET_CONFIG_PEER_IPV4_ADDR=""

--- a/subsys/logging/log_backend_net.c
+++ b/subsys/logging/log_backend_net.c
@@ -130,18 +130,20 @@ static int do_net_init(void)
 
 	} else if (IS_ENABLED(CONFIG_NET_IPV4) &&
 		   server_addr.sa_family == AF_INET) {
-		struct net_if_ipv4 *ipv4;
-		struct net_if *iface;
+		const struct in_addr *src;
 
-		iface = net_if_ipv4_select_src_iface(
-					&net_sin(&server_addr)->sin_addr);
-		ipv4 = iface->config.ip.ipv4;
+		src = net_if_ipv4_select_src_addr(
+				  NULL, &net_sin(&server_addr)->sin_addr);
 
-		net_ipaddr_copy(&local_addr4.sin_addr,
-				&ipv4->unicast[0].address.in_addr);
+		if (src) {
+			net_addr_ntop(AF_INET, src, dev_hostname,
+				      MAX_HOSTNAME_LEN);
 
-		net_addr_ntop(AF_INET, &local_addr4.sin_addr, dev_hostname,
-			      MAX_HOSTNAME_LEN);
+			net_ipaddr_copy(&local_addr4.sin_addr, src);
+		} else {
+			goto unknown;
+		}
+
 	} else {
 	unknown:
 		DBG("Cannot setup local context\n");

